src/lxc/log.c

```diff
@@ -155,8 +155,10 @@ static int log_append_logfile(const struct lxc_log_appender *appender,
 	if (fd_to_use == -1)
 		return 0;
 
-	t = localtime(&event->timestamp.tv_sec);
-	strftime(date, sizeof(date), "%Y%m%d%H%M%S", t);
+	if (!localtime_r(&event->timestamp.tv_sec, &newtime))
```

Comment:
For future reference, this fix is not at all complete. The race actually comes from `localtime()`/`localtime_r()` and friends using `tzset()` internally which seems to deadlock in multi-threaded scenarios. My suspicion is that `fork()`ing on daemonized start in multi-threaded environments messes up the mutexes this function uses internally.